server/src/main/java/org/elasticsearch/common/xcontent/ChunkedToXContentHelper.java

```diff
@@ -73,6 +73,10 @@ public static Iterator<ToXContent> field(String name, long value) {
         return Iterators.single((b, p) -> b.field(name, value));
     }
 
+    public static Iterator<ToXContent> field(String name, double value) {
+        return Iterators.single((b, p) -> b.field(name, value));
```

Comment:
This is not a good idea. If `name` is a constant value then you're now capturing it in a couple places. Since we have virtual callsites here anyway, that just way increases the cost because we don't gain anything from any drying up.

Reply:
OK - what about the other `field` methods? Is it worth flattening all the calls that use a constant `name` into the respective `toXContentChunked` methods?

Reply:
Technically yes. Whether there's practical value in that outside of the search results and bulk results is a different question though. But for the very hot paths it's a clear yes.